gslb/ingestion/gslb_host_rule.go

```diff
@@ -76,43 +76,97 @@ func updateGSLBHR(gslbhr *gslbhralphav1.GSLBHostRule, msg string, status string)
 	}
 }
 
+func ProportionateToSum20(gslbhr *gslbhralphav1.GSLBHostRule, tseSum int) {
+	var factor float64 = float64(20) / float64(tseSum)
+	var sum uint32 = 0
+	var diff uint32
+	for i, _ := range gslbhr.Spec.TrafficSplit {
+		gslbhr.Spec.TrafficSplit[i].Weight = uint32(math.Round(float64(gslbhr.Spec.TrafficSplit[i].Weight) * factor))
+		sum += gslbhr.Spec.TrafficSplit[i].Weight
+	}
+	if sum == 20 {
+		return
+	} else if sum > 20 {
+		diff = sum - 20
+		sort.Slice(gslbhr.Spec.TrafficSplit[:], func(i, j int) bool {
+			return gslbhr.Spec.TrafficSplit[i].Weight > gslbhr.Spec.TrafficSplit[j].Weight
+		})
+		for i := 0; i < len(gslbhr.Spec.TrafficSplit) && diff > 0; i++ {
+			gslbhr.Spec.TrafficSplit[i].Weight -= uint32(1)
+			diff -= 1
+		}
+	} else if sum < 20 {
+		diff = 20 - sum
+		sort.Slice(gslbhr.Spec.TrafficSplit[:], func(i, j int) bool {
+			return gslbhr.Spec.TrafficSplit[i].Weight > gslbhr.Spec.TrafficSplit[j].Weight
+		})
+		for i := 0; i < len(gslbhr.Spec.TrafficSplit) && diff > 1; i++ {
+			gslbhr.Spec.TrafficSplit[i].Weight += uint32(1)
+			diff += 1
+		}
+	}
+}
+
 func isSitePersistenceProfilePresent(gslbhr *gslbhralphav1.GSLBHostRule, profileName string) bool {
 	// Check if the profile mentioned in gslbHostRule are present as application persistence profile on the gslb leader
 	aviClient := avictrl.SharedAviClients().AviClient[0]
-	uri := "/api/applicationpersistenceprofile?name=" + profileName
+	uri := "api/applicationpersistenceprofile"
```

Comment:
Is there a reason we are fetching all the app persistence profiles? Shouldn't just the one profile be enough?